File: java/src/main/java/com/myriad/wdl/model/expressions/WdlObjectLiteral.java
package com.myriad.wdl.model.expressions;

import com.myriad.wdl.model.base.WdlKeyValue.WdlStringKeyValue;
import java.util.ArrayDeque;

public final class WdlObjectLiteral implements WdlExpression {
  private final ArrayDeque<WdlObjectEntry> entries = new ArrayDeque<>();

  public WdlObjectLiteral() {}

  public ArrayDeque<WdlObjectEntry> entries() {
    return entries;
  }

  @Override
  public String toString() {
    return getClass().getSimpleName();
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.OBJ_LIT;
  }

  public static class WdlObjectEntry extends WdlStringKeyValue {
    public WdlObjectEntry() {
      super();
    }

    public WdlObjectEntry(String key, WdlExpression value) {
      super(key, value);
    }

    public WdlObjectEntry(String key) {
      super(key);
    }
  }
}

File: typescript/src/model/expressions/wdl-object-literal.ts
/** Object literal expression nodes. */
import { WdlStringKeyValue } from '../base/wdl-key-value.js';
import { WdlExpressionComponentType, type WdlExpression } from './wdl-expression.js';

/** Single named entry inside an object literal. */
export class WdlObjectEntry extends WdlStringKeyValue {}

/** Models a WDL object literal. */
export class WdlObjectLiteral implements WdlExpression {
  private readonly entryValues: WdlObjectEntry[] = [];

  /** Returns the ordered object entries contained in the literal. */
  public entries(): WdlObjectEntry[] {
    return this.entryValues;
  }

  /** Returns the broad expression family for this node. */
  public componentType(): WdlExpressionComponentType {
    return WdlExpressionComponentType.OBJ_LIT;
  }
}
